pkg/operator/starter.go

```diff
@@ -119,105 +154,138 @@ func RunOperator(ctx context.Context, controllerConfig *controllercmd.Controller
-			nodeInformer.Informer(),
-			cloudConfigInformer.Informer(),
-			infraInformer.Informer(),
-			configMapInformer.Informer(),
+			controlPlaneSecretInformer.Informer(),
+			guestNodeInformer.Informer(),
+			controlPlaneCloudConfigInformer.Informer(),
+			guestInfraInformer.Informer(),
+			controlPlaneConfigMapInformer.Informer(),
 		},
-		csidrivercontrollerservicecontroller.WithSecretHashAnnotationHook(defaultNamespace, secretName, secretInformer),
+		withHypershiftDeploymentHook(isHypershift),
+		withHypershiftReplicasHook(isHypershift, guestNodeInformer.Lister()),
+		withNamespaceDeploymentHook(controlPlaneNamespace),
+		csidrivercontrollerservicecontroller.WithSecretHashAnnotationHook(controlPlaneNamespace, secretName, controlPlaneSecretInformer),
 		csidrivercontrollerservicecontroller.WithObservedProxyDeploymentHook(),
-		csidrivercontrollerservicecontroller.WithReplicasHook(nodeInformer.Lister()),
-		withCustomCABundle(cloudConfigLister),
-		withCustomTags(infraInformer.Lister()),
-		withCustomEndPoint(infraInformer.Lister()),
+		withCustomAWSCABundle(guestInfraInformer.Lister(), controlPlaneCloudConfigLister),
+		withCustomTags(guestInfraInformer.Lister()),
+		withCustomEndPoint(guestInfraInformer.Lister()),
 		csidrivercontrollerservicecontroller.WithCABundleDeploymentHook(
-			defaultNamespace,
+			controlPlaneNamespace,
 			trustedCAConfigMap,
-			configMapInformer,
-		),
-	).WithCSIDriverNodeService(
-		"AWSEBSDriverNodeServiceController",
-		assets.ReadFile,
-		"node.yaml",
-		kubeClient,
-		kubeInformersForNamespaces.InformersFor(defaultNamespace),
-		[]factory.Informer{configMapInformer.Informer()},
-		csidrivernodeservicecontroller.WithObservedProxyDaemonSetHook(),
-		csidrivernodeservicecontroller.WithCABundleDaemonSetHook(
-			defaultNamespace,
-			trustedCAConfigMap,
-			configMapInformer,
+			controlPlaneConfigMapInformer,
 		),
 	).WithServiceMonitorController(
 		"AWSEBSDriverServiceMonitorController",
-		dynamicClient,
-		assets.ReadFile,
+		controlPlaneDynamicClient,
+		assetWithNamespaceFunc(controlPlaneNamespace),
 		"servicemonitor.yaml",
 	).WithStorageClassController(
 		"AWSEBSDriverStorageClassController",
 		assets.ReadFile,
 		"storageclass_gp3.yaml",
-		kubeClient,
-		kubeInformersForNamespaces.InformersFor(""),
+		controlPlaneKubeClient,
+		controlPlaneKubeInformersForNamespaces.InformersFor(""),
 	)
 	if err != nil {
 		return err
 	}
 
-	caSyncController, err := newCustomCABundleSyncer(
-		operatorClient,
-		kubeInformersForNamespaces,
-		kubeClient,
-		controllerConfig.EventRecorder,
+	// Start controllers that manage resources in GUEST clusters.
+	guestCSIControllerSet := csicontrollerset.NewCSIControllerSet(
+		guestOperatorClient,
+		eventRecorder,
+	).WithStaticResourcesController(
+		"AWSEBSDriverGuestStaticResourcesController",
+		guestKubeClient,
+		guestDynamicClient,
+		guestKubeInformersForNamespaces,
+		assets.ReadFile,
+		[]string{
+			"storageclass_gp2.yaml",
+			"storageclass_gp3.yaml",
```

Comment:
storageclass_gp3.yaml should be created by `StorageClassController`

Reply:
Fixed.